File: src/cli/team/list.ts
import type { Command } from "commander";
import { getClient } from "../../lib/client.ts";
import { printError, printPaginated } from "../../lib/output.ts";

export function registerList(team: Command): void {
  team
    .command("list")
    .description("List all teams")
    .option("--limit <n>", "Limit results", "50")
    .option("--cursor <token>", "Pagination cursor for next page")
    .action(async (opts: { limit: string; cursor?: string }) => {
      try {
        const client = getClient();
        const results = await client.teams({
          first: parseInt(opts.limit, 10),
          after: opts.cursor,
        });
        const items = results.nodes.map((t) => ({
          id: t.id,
          name: t.name,
          key: t.key,
        }));
        printPaginated(items, results.pageInfo);
      } catch (err) {
        printError(err instanceof Error ? err.message : "List failed");
      }
    });
}

File: internal/cli/team/list.go
package team

import (
	"context"

	"github.com/spf13/cobra"

	"github.com/shhac/lin/internal/linear"
	"github.com/shhac/lin/internal/output"
)

func registerList(team *cobra.Command) {
	cmd := &cobra.Command{
		Use:   "list",
		Short: "List all teams",
		Args:  cobra.NoArgs,
	}
	page := output.AddPageFlags(cmd)

	cmd.Run = func(_ *cobra.Command, _ []string) {
		client := linear.GetClient()

		resp, err := linear.TeamList(context.Background(), client, nil, page.Size(), page.Cursor())
		if err != nil {
			output.HandleGraphQLError(err)
		}

		items := make([]map[string]any, len(resp.Teams.Nodes))
		for i, t := range resp.Teams.Nodes {
			items[i] = map[string]any{
				"id":   t.Id,
				"name": t.Name,
				"key":  t.Key,
			}
		}

		output.PrintPage(items, resp.Teams.PageInfo.HasNextPage, resp.Teams.PageInfo.EndCursor)
	}

	team.AddCommand(cmd)
}
